packages/embedder/ai_models/TextGenerationInterface.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AbstractGenerationModel,
+  GenerationModelConfig,
+  GenerationModelParams
+} from './AbstractModel'
+
+const MAX_REQUEST_TIME_MS = 120 * 1000
+
+export enum ModelSubTypes {
+  Zephyr_7b = 'TheBloke/zephyr-7b-beta'
+}
+
+const modelSubTypeDefinitions: Record<ModelSubTypes, GenerationModelParams> = {
+  [ModelSubTypes.Zephyr_7b]: {
+    maxInputTokens: 512
+  }
+}
+
+function isValidModelSubType(type: any): type is ModelSubTypes {
+  return Object.values(ModelSubTypes).includes(type)
+}
+
+export class TextGenerationInterface extends AbstractGenerationModel {
+  constructor(config: GenerationModelConfig) {
+    super(config)
+  }
+
+  public async summarize(content: string, temperature: number = 0.8, maxNewTokens: number = 512) {
+    const prompt = `Create a brief, one-paragraph summary of the following: ${content}`
+    const body = {
+      inputs: prompt,
+      parameters: {
+        maxNewTokens: maxNewTokens,
+        temperature: temperature
```

Comment:
+1 this endpoint also accepts a `truncate` param. If we use that, we don't have to do a word count heuristic

Reply:
Oh let me look at that...

Reply:
Implemented truncate on the endpoint, and on node-llama-cpp-text-generation-inference